pyverdict/pyverdict/verdictcontext.py

```diff
@@ -0,0 +1,47 @@
+from .resultset import ResultSet
+import time
+import os
+import subprocess
+import atexit
+import pkg_resources
+from py4j.java_gateway import JavaGateway, launch_gateway, GatewayParameters, java_import
+
+Gateway = None
+
+class VerdictContext:
+    """main interface to interact with the java objects
+    require specifying the jdbc driver jar path
+    """
+
+    def __init__(self, url, usr, pwd, jdbc_driver):
+        init(jdbc_driver)
+        self._context = connect(url, usr, pwd)
+
+    def sql(self, query):
+        return ResultSet(self._context.sql(query))
+
+def init(jdbc_driver):
+    """initialize a py4j gateway
+    starting up a JVM only once
+    """
+
+    global Gateway
```

Comment:
Why do we need a global variable? I like local variables if possible.

Reply:
It's the simplest way I could find to implement a gateway that will only be initialized once. This gateway is shared across all VerdictContext class objects.

Reply:
I see. Let's make the variable as a class variable of VerdictContext.